src/components/Item.js

```diff
@@ -0,0 +1,70 @@
+import PropTypes from 'prop-types';
+import { useState } from 'react';
+
+const Item = ({
+  todo, completedTask, deleteTodo, saved, updateTodos,
+}) => {
+  // set a state for the tod
+  const [currentTodo, setCurrentTodo] = useState(todo);
+  // set initial state for the edit mode
+  const [editMode, setEditMode] = useState(false);
+  // update edited mode to true when the todo about to be edited is clicked
+  const handleEditingMode = () => {
+    setEditMode(true);
+  };
+  // update the edited todos to the list and update the localStorage
+  const updateEdited = (currentTodo) => {
+    const editedTodo = saved().findIndex((t) => t.id === currentTodo.id);
+    const newTodos = saved();
+    newTodos.splice(editedTodo, 1, currentTodo);
+    updateTodos(newTodos);
+  };
+
+  // save edited todo on blur event and prevent enter key from getting triggered
+  const saveEditedTask = (e) => {
+    updateEdited(currentTodo);
```

Comment:
[OPTIONAL CHANGES] - You can remove this commented text, just for the better visual look of the code. :smile:

Reply:
I left them in there to explain my functions better to anyone who views them. Thank you